ext/handlers_http.h

```diff
@@ -178,7 +185,15 @@ static inline void ddtrace_inject_distributed_headers_config(zend_array *array,
                            span_id,
                            sampling_priority > 0);
 
-                zend_string *full_tracestate = ddtrace_format_tracestate(tracestate, origin, sampling_priority, propagated_tags, tracestate_unknown_dd_keys);
+                uint64_t propagated_span_id = 0;
+                zval *old_parent_id;
+                if (root) {
+                    propagated_span_id = span_id;
+                } else if ((old_parent_id = zend_hash_str_find(&DDTRACE_G(root_span_tags_preset), ZEND_STRL("_dd.parent_id")))) {
+                    propagated_span_id = ddtrace_parse_hex_span_id(old_parent_id);
+                }
+
+                zend_string *full_tracestate = ddtrace_format_tracestate(tracestate, propagated_span_id, origin, sampling_priority, propagated_tags, tracestate_unknown_dd_keys);
```

Comment:
`p:<span_id_hex>;` should be set using the current `span_id`.

For example if the php span the following traceparent `00-0000000000000000b5a2814f70060771-7197677932a62370-01` then it's tracestate should contain `p:7197677932a62370`. 

```suggestion
                zend_string *full_tracestate = ddtrace_format_tracestate(tracestate, span_id, origin, sampling_priority, propagated_tags, tracestate_unknown_dd_keys);
```

Reply:
But then the traces will be disconnected, defeating the point of the change in that scenario?

This particular case handles the scenario where a distributed trace is injected, but _no_ active span exists on the local side.

Let's say the injected trace has a trace parent span `1234567890123456` (which is also the locally active span id), and a tracestate `p` of `abcdefabcdefabcd`. If we blindly inject `p:1234567890123456`. Even though a span id `1234567890123456` may never reach datadog.

Reply:
Ahhh I was missing this context. This change looks good to me

Reply:
I'm trying to make sure that I understand this correctly (not super familiar with the PHP tracer) I'm not sure if I fully understand the `else if`:

Essentially if we have an active span we'll inject that span ID as `p`, which makes sense.

But if we are only injecting and propagating context that is what this is attempting to handle correct?
And if so, it'll check for `_dd.parent_id` on the root span and use that value or it'll propagate `0`. I don't think I covered this specific scenario in the spec so just want to make sure I understand what is happening as it may require updating the spec.

Reply:
Yes, that's right. (not quite "on the root span", but the tags which seed the root span whenever one is created, i.e. amongst other things, the extracted tags.)